metrics-operator/api/v1alpha3/analysisvaluetemplate_types.go

```diff
@@ -0,0 +1,59 @@
+/*
+Copyright 2023.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha3
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// AnalysisValueTemplateSpec defines the desired state of AnalysisValueTemplate
+type AnalysisValueTemplateSpec struct {
+	// Provider represents the provider object
+	Provider ProviderRef `json:"provider"`
+	// Query represents the query to be run. It can include placeholders that are defined using the go template
+	// syntax
```

Comment:
would it make sense to include a link to the doc of go templating?
https://pkg.go.dev/text/template